go/vt/vtgate/planbuilder/operators/info_schema_planning.go

```diff
@@ -0,0 +1,314 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operators
+
+import (
+	"strings"
+
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/operators/ops"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+
+	"vitess.io/vitess/go/mysql/collations"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/servenv"
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/engine"
+	"vitess.io/vitess/go/vt/vtgate/evalengine"
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/plancontext"
+)
+
+// InfoSchemaRouting used for information_schema queries.
+// They are special because we usually don't know at plan-time
+// what keyspace the query go to, because we don't see normalized literal values
+type InfoSchemaRouting struct {
+	SysTableTableSchema []sqlparser.Expr
+	SysTableTableName   map[string]sqlparser.Expr
+	Table               *QueryTable
+}
+
+func (isr *InfoSchemaRouting) UpdateRoutingParams(_ *plancontext.PlanningContext, rp *engine.RoutingParameters) error {
+	rp.SysTableTableSchema = nil
+	for _, expr := range isr.SysTableTableSchema {
```

Comment:
now that we do not merge if SysTableTableSchema is different, should SysTableTableSchema be an Expr than slice of Expr? Similarly, Do we need SysTableTableName as a map?

Reply:
We should add a task for it